aggregate_root/lib/aggregate_root/instrumented_repository.rb

```diff
@@ -0,0 +1,33 @@
+module AggregateRoot
+  class InstrumentedRepository
+    def initialize(repository, instrumentation)
+      @repository = repository
+      @instrumentation = instrumentation
+    end
+
+    def load(aggregate, stream_name)
+      instrumentation.instrument("load.repository.aggregate_root",
+                                 aggregate_class: aggregate.class,
+                                 stream_name: stream_name) do
+        repository.load(aggregate, stream_name)
+      end
+    end
+
+    def store(aggregate, stream_name)
+      instrumentation.instrument("store.repository.aggregate_root",
+                                 aggregate_class: aggregate.class,
+                                 aggregate_version: aggregate.version,
+                                 stored_events: aggregate.unpublished_events.size,
```

Comment:
Perhaps we should pass just `aggregate.unpublished_events` here. That would be in symmetry with `append_to_stream.repository.rails_event_store`.